src/main/scala/dxWDL/compiler/Native.scala

```diff
@@ -1545,17 +1567,32 @@ case class Native(dxWDLrtId: Option[String],
             }
             accu + (apl.name -> execRecord)
           case wf: IR.Workflow =>
-            val dxwfl = buildWorkflowIfNeeded(wf, accu)
-            accu + (wf.name -> Native.ExecRecord(wf, dxwfl, Vector.empty))
+            if (wf == bundle.primaryCallable.get) {
```

Comment:
I suspect this will throw an exception when this is a sub-workflow. I am also not sure why we need this clause.